app/assets/javascripts/components/uploads/upload.jsx

```diff
@@ -1,52 +1,74 @@
 import React from 'react';
+import createReactClass from 'create-react-class';
 import PropTypes from 'prop-types';
 
-const Upload = ({ upload, linkUsername }) => {
-  const fileName = upload.file_name;
-  let imageFile;
-  if (upload.deleted) {
-    imageFile = '/assets/images/deleted_image.svg';
-  } else {
-    imageFile = upload.thumburl;
-  }
-
-  let uploadDivStyle = {};
-  if (upload.thumbwidth && upload.thumbheight) {
-    uploadDivStyle = {
-      width: upload.thumbwidth * 250 / upload.thumbheight,
-      flexGrow: upload.thumbwidth * 250 / upload.thumbheight
+const Upload = createReactClass({
+  displayName: 'Upload',
+
+  propTypes: {
+    upload: PropTypes.object,
+    linkUsername: PropTypes.bool
+  },
+
+  getInitialState() {
+    return {
+      width: null,
+      height: null
+    };
+  },
+
+  componentWillMount() {
```

Comment:
Yeah,I saw it was replaced by  `UNSAFE_componentWillMount`.I used `componentDidMount` and works fine with it but we are using `componentWillMount`  and `componentWillReceiveProps` (which was replaced as well)  in a lot of other components. So we won't be manually replacing each so I'd let it be for now because I guess it makes more sense that the styles are set before the component is mounted. What do you say? :sweat_smile:

Reply:
Good to know that it works with componentDidMount. We might as well write new code in ways that will be compatible with React 16+, but it's not a big deal at this point at all. I'll go ahead and merge this.